File: 1465. Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts/Solution.java
import java.util.Arrays;

class Solution {
    private int getMax(int n, int[] cuts) {
        Arrays.sort(cuts);
        int res = Math.max(cuts[0], n - cuts[cuts.length - 1]);
        for (int i = 1; i < cuts.length; i++) {
            res = Math.max(res, cuts[i] - cuts[i - 1]);
        }
        return res;
    }

    public int maxArea(int h, int w, int[] horizontalCuts, int[] verticalCuts) {
        final int P = 1000000007;
        return (int) ((long) getMax(h, horizontalCuts) * getMax(w, verticalCuts) % P);
    }
}

File: 1465. Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts/Solution.cpp
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
    int maxArea(int h, int w, vector<int>& horizontalCuts, vector<int>& verticalCuts)
    {
        const int P = 1000000007;
        return static_cast<long long>(getMax(h, horizontalCuts)) * getMax(w, verticalCuts) % P;
    }

private:
    int getMax(int n, vector<int> &cuts) {
        sort(cuts.begin(), cuts.end());
        int res = max(cuts[0], n - cuts.back());
        for (size_t i = 1; i < cuts.size(); ++i)
            res = max(res, cuts[i] - cuts[i - 1]);
        return res;
    }
};
